src/csharp/Microsoft.Spark.E2ETest/IpcTests/JvmThreadPoolTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Sql;
+using Xunit;
+
+namespace Microsoft.Spark.E2ETest.IpcTests
+{
+    [Collection("Spark E2E Tests")]
+    public class JvmThreadPoolTests
+    {
+        private readonly SparkSession _spark;
+        private readonly IJvmBridge _jvmBridge;
+
+        public JvmThreadPoolTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+            _jvmBridge = ((IJvmObjectReferenceProvider)_spark).Reference.Jvm;
+        }
+
+        /// <summary>
+        /// Test that the active SparkSession is thread-specific.
+        /// </summary>
+        [Fact]
+        public void TestThreadLocalSessions()
+        {
+            SparkSession.ClearActiveSession();
+
+            void testChildThread(string appName)
+            {
+                var thread = new Thread(() =>
+                {
+                    Assert.Null(SparkSession.GetActiveSession());
+
+                    SparkSession.SetActiveSession(
+                        SparkSession.Builder().AppName(appName).GetOrCreate());
+
+                    // Since we are in the child thread, GetActiveSession() should return the child
+                    // SparkSession.
+                    var activeSession = SparkSession.GetActiveSession();
+                    Assert.NotNull(activeSession);
+                    Assert.Equal(appName, activeSession.Conf().Get("spark.app.name", null));
+                });
+
+                thread.Start();
+                while (thread.IsAlive)
+                {
+                    Thread.Sleep(1000);
+                }
```

Comment:
can you just `thread.Join();`?

Reply:
Fixed.

Reply:
Looks like you removed the thread.Start();

Reply:
Oops.  Fixed.